lfucg-exactions/server/plats/views.py

```diff
@@ -28,129 +28,136 @@ def get(self, request, *args, **kwargs):
-        lot_queryset = Lot.objects.filter(plat=plat)
-
-        if lot_queryset is not None:
-            headers.extend([
-                'Address',
-                'Lot Number',
-                'Parcel ID',
-                'Permit ID', 
-                'Latitude',
-                'Longitude',
-                'Roads',
-                'Parks',
-                'Storm Water',
-                'Open Spaces',
-                'Sewer Cap.',
-                'Sewer Trans.',
-                'Total Exactions',
-                'Sewer Exactions',
-                'Non-Sewer Exactions',
-                'Current  Exactions',
-                'Current Sewer Exactions',
-                'Current Non-Sewer Exactions',
-            ])
-        
-            payment_length_per_lot = 0
-            ledger_length_per_lot = 0
-
-            for single_lot in lot_queryset:
-                payment_queryset = Payment.objects.filter(lot_id=single_lot.id)
-                while len(payment_queryset) > payment_length_per_lot:
-                    headers.extend(['Payment Total',])
-                    payment_length_per_lot = payment_length_per_lot + 1
-
-            for single_lot in lot_queryset:
-                ledger_from_queryset = AccountLedger.objects.filter(account_from=single_lot.account)
-                while len(ledger_from_queryset) > ledger_length_per_lot:
-                    headers.extend(['Account Ledger Credits Spent'])
-                    ledger_length_per_lot = ledger_length_per_lot + 1
-
-            writer.writerow(headers)
-
-            for single_lot in lot_queryset:
-                current_lot_data = []
-
-                current_lot_data.extend([
-                    single_lot.address_full,
-                    single_lot.lot_number,
-                    single_lot.parcel_id,
-                    single_lot.permit_id,
-                    single_lot.latitude,
-                    single_lot.longitude,
-                    single_lot.dues_roads_dev,
-                    single_lot.dues_parks_dev,
-                    single_lot.dues_storm_dev,
-                    single_lot.dues_open_space_dev,
-                    single_lot.dues_sewer_cap_dev,
-                    single_lot.dues_sewer_trans_dev,
-                ])
-
-                lot_balance = calculate_lot_balance(single_lot.id)
-
-                current_lot_data.extend([
-                    '${:,.2f}'.format(lot_balance['total_exactions']),
-                    '${:,.2f}'.format(lot_balance['sewer_exactions']),
-                    '${:,.2f}'.format(lot_balance['non_sewer_exactions']),
-                    '${:,.2f}'.format(lot_balance['current_exactions']),
-                    '${:,.2f}'.format(lot_balance['sewer_due']),
-                    '${:,.2f}'.format(lot_balance['non_sewer_due']),
+        try:
+            plat_queryset = Plat.objects.filter(id=plat)
+            
+            plat_object = plat_queryset[0]
+            # if plat_queryset is not None:
```

Comment:
This will work okay, but for best practice it is better to avoid an exception rather than catch one in a try/catch.  I might be missing something, but I think you could probably handle this a little cleaner with:

if plat_queryset is not None:
  plat_queryset = plat_queryset[0]
else:
  return HttpResponse('No Plat was found.')